keras/utils/np_utils.py

```diff
@@ -21,15 +23,24 @@ def to_categorical(y, num_classes=None):
     """
     y = np.array(y, dtype='int')
     input_shape = y.shape
-    if input_shape and input_shape[-1] == 1 and len(input_shape) > 1:
-        input_shape = tuple(input_shape[:-1])
+    channels_axis = 0 if K.image_data_format() == 'channels_first' else -1
```

Comment:
0 is the batch axis, not the channels axis.

Reply:
@fchollet Hmmm, that actually brings up an issue I hadn't thought of, because `to_categorical` may be called either before or after batching samples together. With `channels_last` the channels just go in axis -1 either way, but with `channels_first` they would go in axis 0 if called before adding the batches channel (as I did) and in axis 1 after.

I could add an optional input specifying the axis in which to put the channels. The default could be to put them in axis 1 for `channels_first` and axis -1 for `channels_last`. Does that sound good?

Reply:
On second thought, I don't think we should support this argument at the level of `to_categorical`. Users are responsible for transposing their Numpy data before feeding it into models (as seen in all Keras example scripts, which are almost always compatible with both data formats).

The global `data_format` is a Keras thing, a model-level thing, but preprocessing utilities (of which this is one, conceptually) are not really part of the Keras API per se. Also, it's weird to have a dependency on the backend in `np_utils`.

Reply:
@fchollet Fair enough. So should I just document in the header comment that the channels are placed last? Or not even that?

Reply:
Yes, you can add a comment like that.

Reply:
@fchollet Or maybe I should just add an optional input specifying the axis in which to put the channels, where the default axis is always -1, regardless of the Keras `image_data_format`?